airbyte-cdk/java/airbyte-cdk/db-destinations/src/main/java/io/airbyte/cdk/integrations/destination/jdbc/typing_deduping/JdbcDestinationHandler.java

```diff
@@ -59,7 +65,49 @@ public boolean isFinalTableEmpty(final StreamId id) throws Exception {
 
   @Override
   public InitialRawTableState getInitialRawTableState(final StreamId id) throws Exception {
-    return new InitialRawTableState(true, Optional.empty());
+    final ResultSet tables = jdbcDatabase.getMetaData().getTables(
+        databaseName,
+        id.rawNamespace(),
+        id.rawName(),
+        null);
+    if (!tables.next()) {
+      // There's no raw table at all. Therefore there are no unprocessed raw records, and this sync
+      // should not filter raw records by timestamp.
+      return new InitialRawTableState(false, Optional.empty());
+    }
+    // And use two explicit queries because COALESCE might not short-circuit evaluation.
+    // This first query tries to find the oldest raw record with loaded_at = NULL.
+    // Unsafe query requires us to explicitly close the Stream, which is inconvenient,
+    // but it's also the only method in the JdbcDatabase interface to return non-string/int types
+    try (final Stream<Timestamp> timestampStream = jdbcDatabase.unsafeQuery(
+        conn -> conn.prepareStatement(
+            DSL.select(DSL.field("MIN(_airbyte_extracted_at)").as("min_timestamp"))
+                .from(DSL.name(id.rawNamespace(), id.rawName()))
+                .where(DSL.condition("_airbyte_loaded_at IS NULL"))
+                .getSQL()),
+        record -> record.getTimestamp("min_timestamp"))) {
+      // Filter for nonNull values in case the query returned NULL (i.e. no unloaded records).
+      final Optional<Timestamp> minUnloadedTimestamp = timestampStream.filter(Objects::nonNull).findFirst();
+      if (minUnloadedTimestamp.isPresent()) {
+        // Decrement by 1 second since timestamp precision varies between databases.
+        final Optional<Instant> ts = minUnloadedTimestamp
+            .map(Timestamp::toInstant)
+            .map(i -> i.minus(1, ChronoUnit.SECONDS));
```

Comment:
this is slightly different from the original redshift implementation - instead of doing a timestamp subtraction within redshift, we just query the raw timestamp and decrement it in java code. Timestamp math can be dialect-specific, so probably not a good idea to have it in a base Jdbc implementation.